src/modules/ethereum/transactions/gasPrices/optimistic/fee.ts

```diff
@@ -69,7 +93,11 @@ export async function createOptimisticFee({
 
   const l1_fee_est = Math.round(l1 * l1GasEstimation);
   return {
-    maxFee: l2 * Number(gas) + Math.round(l1_fee_est * 1.25), // Optimism guarantees that there should not be one-time jumps of l1 component of gas estimation more than 25%
-    estimatedFee: l2 * Number(gas) + l1_fee_est,
+    maxFee:
+      (maxFeePerGas ? Number(maxFeePerGas) : Number(gasPrice)) * Number(gas) +
+      Math.round(l1_fee_est * 1.25), // Optimism guarantees that there should not be one-time jumps of l1 component of gas estimation more than 25%
+    estimatedFee:
+      (maxFeePerGas ? Number(maxFeePerGas) : Number(gasPrice)) * Number(gas) +
```

Comment:
I really didn't want to import base fee here from eip1559 just for this case.

Reply:
In this case estimated fee will be too off. maxFeePerGas is usually x2 compared to baseFee

Reply:
Tryed to fix